### PR TITLE
XML-RPC: Fixup `modify_setting` with boolean values

### LINKED ISSUE #3336: XML-RPC: Boolean values cannot be set with `modify_setting`
### Describe the bug

```
[root@localhost cobbler]# cobbler sync --dhcp
task started: 2023-02-02_051336_Sync_f09f71089bbe47bc8299204141feebf2
task started (id=Sync, time=Thu Feb  2 05:13:36 2023)
start_task(sync); event_id(2023-02-02_051336_Sync_f09f71089bbe47bc8299204141feebf2)
"manage_dhcp" not set. Skipping DHCP sync.
*** TASK COMPLETE ***

[root@localhost cobbler]# cobbler setting edit --name=manage_dhcp --value=true
Changing the setting failed
```

The configuration file was not changed.

`settings.yaml`:

```yaml
manage_dhcp: false
manage_dhcp_v4: false
```

### Expected behavior

The configuration file can be written normally

### Cobbler version

_v3.4.0_

### Operating system

```
[root@localhost dhcp]# cat /etc/os-release 
NAME="Fedora Linux"
VERSION="37 (Server Edition)"
ID=fedora
VERSION_ID=37
```
### Cobbler log

```
INFO | REMOTE create_task(Sync); event_id(2023-02-02_051425_Sync_cc59548baaf74238851c757a040bd3a7); user(?)
INFO | start_task(sync); event_id(2023-02-02_051425_Sync_cc59548baaf74238851c757a040bd3a7)
INFO | "manage_dhcp" not set. Skipping DHCP sync.
INFO | ### TASK COMPLETE ###
```

### Screenshots

None

### PR DESCRIPTION
**Pull request by xuezhixin**
## Linked Items

Fixes #3336

 

## Description
type(isinstance(getattr(self.api.settings(), 'manage_dhcp')) is bool

isinstance(False,int) is true

## Behaviour changes

Old: 

New: 

## Category

This is related to a:

- [x] Bugfix
- [ ] Feature
- [ ] Packaging
- [ ] Docs
- [ ] Code Quality
- [ ] Refactoring
- [ ] Miscellaneous

## Tests

- [ ] Unit-Tests were created
- [ ] System-Tests were created
- [ ] Code is already covered by Unit-Tests
- [ ] Code is already covered by System-Tests
- [ ] No tests required

### CODE REVIEW

**Comment by SchoolGuy** [comment]
@xuezhixin Many thanks for finding the bug and fixing it right away! Can you show with a pytest testcase that this is working now as intended please?

**Comment by xuezhixin** [comment]
OK, I just resubmitted a commit and performed a function test locally

```
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.11.0, pytest-7.1.3, pluggy-1.0.0
rootdir: /root/cobbler, configfile: pytest.ini
collected 3 items                                                                                                                                                                                                 

tests/actions/sync_test.py sss                                                                                                                                                                              [100%]

================================================================================================ warnings summary =================================================================================================
../../usr/lib64/python3.11/site-packages/Cheetah/Template.py:24
  /usr/lib64/python3.11/site-packages/Cheetah/Template.py:24: DeprecationWarning: 'cgi' is deprecated and slated for removal in Python 3.13
    import cgi  # Used by .webInput() if the template is a CGI script.

cobbler/actions/reposync.py:13
  /root/cobbler/cobbler/actions/reposync.py:13: DeprecationWarning: 'pipes' is deprecated and slated for removal in Python 3.13
    import pipes

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
========================================================================================= 3 skipped, 2 warnings in 0.03s ==========================================================================================
```

`[root@localhost rpmbuild]# cobbler setting edit --name=manage_dhcp_v4 --value=true
[root@localhost rpmbuild]# cobbler setting edit --name=manage_dhcp --value=true
[root@localhost rpmbuild]# cobbler sync --dhcp
task started: 2023-02-03_013341_Sync_597cdbbd12ab4d498e5f04839164af87
task started (id=Sync, time=Fri Feb  3 01:33:41 2023)
start_task(sync); event_id(2023-02-03_013341_Sync_597cdbbd12ab4d498e5f04839164af87)
sync_dhcp
generating /etc/dhcp/dhcpd.conf
running: ['/usr/sbin/dhcpd', '-t', '-q']
received on stdout: 
running: ['systemctl', 'restart', 'dhcpd']
received on stdout: 
*** TASK COMPLETE ***
[root@localhost rpmbuild]# 
`

**Comment by SchoolGuy** [comment]
@xuezhixin The tests are running automatically. I am interested that you add a test case in our tests directory at the appropriate location that shows that there was a bug before and now with your commit it is fixed.

**Comment by xuezhixin** [comment]
OK, I will try to write pytest locally and test it

**Comment by xuezhixin** [comment]
```
[root@localhost tests]# vim ../cobbler/remote.py 
[root@localhost tests]# pytest settings/modify_setting_manage_dhcp_test.py 
========================================= test session starts ==========================================
platform linux -- Python 3.11.0, pytest-7.1.3, pluggy-1.0.0
rootdir: /root/cobbler, configfile: pytest.ini
collected 1 item                                                                                       

settings/modify_setting_manage_dhcp_test.py F                                                    [100%]

=============================================== FAILURES ===============================================
__________________________ TestMiscellaneous.test_modify_setting_manage_dhcp ___________________________

self = <tests.settings.modify_setting_manage_dhcp_test.TestMiscellaneous object at 0x7fb94499a1d0>
remote = <cobbler.remote.CobblerXMLRPCInterface object at 0x7fb944892e50>
token = 'Ve/P3dElNSdqobP6Jl9yOuR5+OHcHU+k2g=='

    def test_modify_setting_manage_dhcp(self, remote, token):
        """
        Test configuration manage_dhcp does not work
        """
        # Assert
        result = remote.modify_setting("manage_dhcp",'true' , token)
>       assert result == 0
E       assert 1 == 0

settings/modify_setting_manage_dhcp_test.py:62: AssertionError
=========================================== warnings summary ===========================================
../../../usr/lib64/python3.11/site-packages/Cheetah/Template.py:24
  /usr/lib64/python3.11/site-packages/Cheetah/Template.py:24: DeprecationWarning: 'cgi' is deprecated and slated for removal in Python 3.13
    import cgi  # Used by .webInput() if the template is a CGI script.

../cobbler/actions/reposync.py:13
  /root/cobbler/cobbler/actions/reposync.py:13: DeprecationWarning: 'pipes' is deprecated and slated for removal in Python 3.13
    import pipes

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
======================================= short test summary info ========================================
FAILED settings/modify_setting_manage_dhcp_test.py::TestMiscellaneous::test_modify_setting_manage_dhcp
==================================== 1 failed, 2 warnings in 0.73s =====================================
[root@localhost tests]# vim ../cobbler/remote.py 
[root@localhost tests]# pytest settings/modify_setting_manage_dhcp_test.py 
========================================= test session starts ==========================================
platform linux -- Python 3.11.0, pytest-7.1.3, pluggy-1.0.0
rootdir: /root/cobbler, configfile: pytest.ini
collected 1 item                                                                                       

settings/modify_setting_manage_dhcp_test.py .                                                    [100%]

=========================================== warnings summary ===========================================
../../../usr/lib64/python3.11/site-packages/Cheetah/Template.py:24
  /usr/lib64/python3.11/site-packages/Cheetah/Template.py:24: DeprecationWarning: 'cgi' is deprecated and slated for removal in Python 3.13
    import cgi  # Used by .webInput() if the template is a CGI script.

../cobbler/actions/reposync.py:13
  /root/cobbler/cobbler/actions/reposync.py:13: DeprecationWarning: 'pipes' is deprecated and slated for removal in Python 3.13
    import pipes

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
==================================== 1 passed, 2 warnings in 0.91s =====================================
[root@localhost tests]# 


```
I try to write tests and test the modified code and the modified code.

File: tests/settings/modify_setting_manage_dhcp_test.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        result = remote.modify_setting("manage_dhcp",'true' , token)
+        assert result == 0
+    
+    def test_modify_setting_bool_parameter(self, remote, token):
+        """
+        Test whether bool as a parameter will report an error
+        """
+        # Assert
+        result = isinstance(False, int)
+        assert result == True
```
Comment (SchoolGuy) [inline tests/settings/modify_setting_manage_dhcp_test.py:70]: This test is not needed. Thus let's remove it. It is only validating the guarantees of Python. This is outside the scope of what we want to test.

```suggestion
```

File: tests/settings/modify_setting_manage_dhcp_test.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    """
+    Initialises the api object and makes it available to the test.
+    """
+    # create XML-RPC client and connect to server
+    remote = CobblerXMLRPCInterface(cobbler_api)
+    shared_secret = get_shared_secret()
+    token = remote.login("", shared_secret)
+    if not token:
+        sys.exit(1)
+    return remote, token
```
Comment (SchoolGuy) [inline tests/settings/modify_setting_manage_dhcp_test.py:52]: These fixtures are a redefinition of the XML-RPC fixtures. Please don't duplicate them.

```suggestion
```

File: tests/settings/modify_setting_manage_dhcp_test.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+
+class TestMiscellaneous:
+    def test_modify_setting_manage_dhcp(self, remote, token):
+        """
+        Test configuration manage_dhcp does not work
+        """
+        # Assert
+        result = remote.modify_setting("manage_dhcp",'true' , token)
+        assert result == 0
```
Comment (SchoolGuy) [inline tests/settings/modify_setting_manage_dhcp_test.py:62]: A test class is not needed as there is no benefit for setup and teardown. Thus let's not use this superfluous wrapper.

```suggestion
def test_modify_setting_manage_dhcp(remote, token):
    """
    Test configuration manage_dhcp does not work
    """
    # Assert
    result = remote.modify_setting("manage_dhcp",'true' , token)
    assert result == 0
```

File: tests/settings/modify_setting_manage_dhcp_test.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
+import pytest
```
Comment (SchoolGuy) [inline tests/settings/modify_setting_manage_dhcp_test.py:1]: Please move this file into the XML-RPC test directory. It doesn't validate the settings directly but only the XML-RPC interface that we use.

**Comment by xuezhixin** [comment]
Thank you very much for your suggestion. I will revise and submit pytest again. In addition, I don't understand that the XML-RPC test directory test pytest will be set to allow by default_ dynamic_ Is Settings * * true * *, because we need to pass here to conduct the subsequent test api.

**Comment by SchoolGuy** [comment]
@xuezhixin I will push a patch to your branch to show you how to achieve that.

**Review by SchoolGuy — APPROVED** [review]
LGTM. Thanks a lot for your work and participation @xuezhixin. In case you have questions or need further help, feel free to open new bugs or GitHub discussions.
